Mathlib/NumberTheory/Padics/PadicVal.lean

```diff
@@ -535,6 +583,24 @@ theorem padicValNat_primes {q : ℕ} [hp : Fact p.Prime] [hq : Fact q.Prime] (ne
     (not_congr (Iff.symm (prime_dvd_prime_iff_eq hp.1 hq.1))).mp neq
 #align padic_val_nat_primes padicValNat_primes
 
+/-- The p-adic valuation of `n` is less than or equal to its logarithm w.r.t `p`.-/
+lemma le_nat_log (n : ℕ) [hp : Fact (Nat.Prime p)]: padicValNat p n ≤ Nat.log p n := by
+  by_cases (n = 0)
+  · simp only [h, padicValNat.zero, Nat.log_zero_right, le_refl]
+  · refine' Nat.le_log_of_pow_le (Nat.Prime.one_lt hp.elim) _
+    by_contra Hnot; push_neg at Hnot
+    exact h (Nat.eq_zero_of_dvd_of_lt (pow_padicValNat_dvd (p := p) (n := n)) Hnot)
+
+lemma le_nat_log_gen {n₁ n₂ : ℕ} [Fact (Nat.Prime p)] (hn : n₁ ≤ n₂):
```

Comment:
```suggestion
lemma le_nat_log_gen {n₁ n₂ : ℕ} [Fact (Nat.Prime p)] (hn : n₁ ≤ n₂) :
```

Reply:
Also what does `gen` mean here?

Reply:
Err. Poor naming. I renamed it to `le_nat_log_of_le`